### PR TITLE
ELPP-2339 Add IIIF images

### PR DESCRIPTION
**Pull request by thewilkybarkid**

### CODE REVIEW

File: src/eLife/Medium/Kernel.php [hunk 1]
```diff
@@ -263,9 +266,28 @@ public static function import(Application $app, string $mediumUsername)
...
-            if (null === $databaseArticle) {
-                $article->save();
+            $imageFile = self::IMAGE_FOLDER.'/'.$article->getImagePath();
+            if (!is_file($imageFile) && !empty($article->getImagePath())) {
+                $file = try_fopen($imageFile, 'wb');
+
+                try {
+                    $image = $app['medium.client']->get("https://{$article->getImageDomain()}/{$article->getImagePath()}", ['sink' => $file]);
+                    if (!in_array($image->getHeaderLine('Content-Type'), ['image/jpeg', 'image/png'])) {
+                        @unlink($imageFile);
```
Comment (giorgiosironi) [inline src/eLife/Medium/Kernel.php]: brrr.. `if (file_exists($imageFile)) { ... }`?

File: src/eLife/Medium/MediumArticleMapper.php [hunk 2]
```diff
@@ -33,11 +35,13 @@ public static function map(callable $fn, $items)
     public static function mapResponseFromMediumArticle(MediumArticle $article) : MediumArticleResponse
     {
         $image = null;
-        if ($article->getImageDomain() && $article->getImagePath()) {
+        if ($article->getImageDomain() && $article->getImagePath() && $article->getImageMediaType()) {
             $image = new ImageResponse(
-            $article->getImageAlt(),
-            Image::basic($article->getImageDomain(), $article->getImagePath())
-          );
+                "https://iiif.elifesciences.org/medium:{$article->getImagePath()}",
```
Comment (giorgiosironi) [inline src/eLife/Medium/MediumArticleMapper.php]: needs to be parameterized by env

Comment (thewilkybarkid) [reply src/eLife/Medium/MediumArticleMapper.php]: Definitely. Looks like we only have a Propel config file at the moment, trying to work out if we can replace that with a generic one.

**Comment by thewilkybarkid** [comment]
This will populate images for items currently in the RSS feed, but won't for previous items. I think it's fine to ignore them (while we could write a command to fetch the images etc, we don't need them at the moment).

**Comment by thewilkybarkid** [comment]
Needs a formula update for the tests to pass.

File: src/eLife/Medium/MediumArticleMapper.php [hunk 3]
```diff
@@ -13,31 +15,33 @@
 {
     const API_VERSION = '1';
 
-    public static function mapResponseFromDatabaseResult($articles) : MediumArticleListResponse
+    public static function mapResponseFromDatabaseResult($articles, string $iiif) : MediumArticleListResponse
```
Comment (thewilkybarkid) [inline src/eLife/Medium/MediumArticleMapper.php:18]: This is an architecture problem, but I don't want a large-scale refactor now.

File: config.example.php [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<?php
+
+return [
+    'iiif' => 'https://iiif.elifesciences.org/medium:',
```
Comment (giorgiosironi) [inline config.example.php:4]: needs to be added to https://github.com/elifesciences/builder-private/blob/master/pillar/iiif.sls#L16

File: src/eLife/Medium/Kernel.php [hunk 5]
```diff
@@ -20,25 +20,31 @@
...
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\HttpKernel\Exception\HttpExceptionInterface;
 use Throwable;
+use function GuzzleHttp\Psr7\try_fopen;
 
 final class Kernel
 {
     const ROOT = __DIR__.'/../../..';
+    const IMAGE_FOLDER = self::ROOT.'/web/images';
```
Comment (giorgiosironi) [inline src/eLife/Medium/Kernel.php:38]: indeed this should be created by the formula to be writable

File: src/eLife/Medium/Kernel.php [hunk 6]
```diff
@@ -20,25 +20,31 @@
...
     const PROPEL_CONFIG = self::ROOT.'/var/cache/propel/conf/config.php';
+    const CONFIG = self::ROOT.'/config.php';
```
Comment (giorgiosironi) [inline src/eLife/Medium/Kernel.php:40]: same for `config.php` probably

File: src/eLife/Medium/Kernel.php [hunk 7]
```diff
@@ -263,9 +273,28 @@ public static function import(Application $app, string $mediumUsername)
         $articles = MediumArticleMapper::xmlToMediumArticleList($data->getBody());
         $responseArticles = [];
         foreach ($articles as $k => $article) {
-            $databaseArticle = (new MediumArticleQuery())->findOneByGuid($article->getGuid());
-            if (null === $databaseArticle) {
-                $article->save();
+            $imageFile = self::IMAGE_FOLDER.'/'.$article->getImagePath();
```
Comment (giorgiosironi) [inline src/eLife/Medium/Kernel.php:278]: does this path ever contain `/`?

**Review by giorgiosironi — APPROVED** [review]

**Comment by giorgiosironi** [comment]
Some comments on what to put in the formula

**Pull request merged**